src/shogun/classifier/NearestCentroid.cpp

```diff
@@ -88,7 +99,28 @@ namespace shogun{
 		m_is_trained=true;
 		distance->init(centroids_feats, distance->get_rhs());
 
-		return true;
+		return shared_from_this()->as<NearestCentroid>();
 	}
 
+	bool NearestCentroid::train_machine(std::shared_ptr<Features> data)
+	{
+		error("the train_machine have been deprecated, please use fit instead");
+		return false;
+	}
+
+	std::shared_ptr<Labels>
+	NearestCentroid::predict(std::shared_ptr<Features> data)
+	{
+		require(data, "features not set");
```

Comment:
@vigsterkr @gf712 while we are touching these things. Wouldnt it be better to do all these sort of input checks in the base class and then call a (virtual) method that can assume that
* the pointers are not null
* the number of examples are the same

so we don't have to have all this duplicate checks?

Reply:
yup agreed, like `train` calls `train_machine`, `predict` does the checks and `predict_labels` is pure virtual